workspace/__ardulib__/XColorPattern/XColorPattern.h

```diff
@@ -0,0 +1,46 @@
+#ifndef __X_COLOR_PATTERN_H__
+#define __X_COLOR_PATTERN_H__
+
+using xod::XColor;
+
+struct PatternNode {
+    XColor color;
+    PatternNode* _next = NULL;
```

Comment:
Is it a struct (that is, all fields are public and dumb by design)? Or is it a class (members are hidden, the interface exposed through methods)?

It’s an inconsistent mix now. Please, stick to a single camp. Looking at the `Pattern` I recommend to follow the “class” way.